### PR TITLE
[TE] Expand config env-var test coverage

### PR DESCRIPTION
**Pull request by jfeng18**
## Description

Expand `config_test.cpp` from 7 tests (MC_PKEY_INDEX only) to 30 tests covering 7 additional env-var parsing patterns and the `ValidatePortRange` utility function.

New test fixtures:
- **MC_NUM_CQ_PER_CTX**: range [1,255], verifies JFC/JFCE side-assignment
- **MC_SLICE_SIZE**: atoi to size_t, zero rejection
- **MC_LOG_LEVEL**: string enum (TRACE/INFO/WARNING/ERROR), trace flag
- **MC_DISABLE_METACACHE**: presence-based boolean
- **MC_IB_TC**: stoi with try-catch, range [0,255], non-numeric rejection
- **MC_ENDPOINT_STORE_TYPE**: string enum (FIFO/SIEVE)
- **ValidatePortRange**: 7 cases covering valid ranges, min>max, well-known ports (0-1023), ephemeral ports (32768-60999), boundary values

## Module

- [x] Transfer Engine (`mooncake-transfer-engine`)

## Type of Change

- [x] Other (test coverage expansion)

## How Has This Been Tested?

- Tests follow the existing `PkeyIndexEnvTest` pattern (setenv/unsetenv in fixture)
- Each assertion verified against config.cpp source code
- CI will run via `ctest` in the existing test suite

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have added tests to prove my changes are effective.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: mooncake-transfer-engine/tests/config_test.cpp [hunk 1]
```diff
@@ -79,5 +81,253 @@ TEST_F(PkeyIndexEnvTest, EmptyStringKeepsDefault) {
...
+    EXPECT_EQ(config.slice_size, 131072u);
+}
+
+TEST_F(SliceSizeEnvTest, ZeroIsRejected) {
+    ASSERT_EQ(::setenv("MC_SLICE_SIZE", "0", 1), 0);
+    GlobalConfig config;
+    config.slice_size = 65536;
+    loadGlobalConfig(config);
+    EXPECT_EQ(config.slice_size, 65536u);
+}
```
[the remark this answers is not in the document]
Comment (jfeng18) [reply mooncake-transfer-engine/tests/config_test.cpp:578]: Good catch — `size_t val = atoi("-100")` wraps to a large value passing `val > 0`. This is a pre-existing config.cpp bug (same class as the EFA stoull issue). Adding the negative test would expose it, but fixing the bug is out of scope for this test PR. Will add the test as a discriminating signal.

File: mooncake-transfer-engine/tests/config_test.cpp [hunk 2]
```diff
@@ -79,5 +81,253 @@ TEST_F(PkeyIndexEnvTest, EmptyStringKeepsDefault) {
...
+    loadGlobalConfig(config);
+    EXPECT_EQ(config.endpoint_store_type, EndpointStoreType::FIFO);
+}
+
+TEST_F(EndpointStoreTypeEnvTest, InvalidIsRejected) {
+    ASSERT_EQ(::setenv("MC_ENDPOINT_STORE_TYPE", "LRU", 1), 0);
+    GlobalConfig config;
+    loadGlobalConfig(config);
+    EXPECT_EQ(config.endpoint_store_type, EndpointStoreType::SIEVE);
+}
```
[the remark this answers is not in the document]
Comment (jfeng18) [reply mooncake-transfer-engine/tests/config_test.cpp:727]: Fair point. Will set sentinel to FIFO before testing invalid value to distinguish preservation from default.

File: mooncake-transfer-engine/tests/config_test.cpp [hunk 3]
```diff
@@ -79,5 +81,253 @@ TEST_F(PkeyIndexEnvTest, EmptyStringKeepsDefault) {
...
+    auto [min_p, max_p] = ValidatePortRange(32768, 40000, 15000, 17000);
+    EXPECT_EQ(min_p, 15000);
+    EXPECT_EQ(max_p, 17000);
+}
+
+TEST(ValidatePortRangeTest, BoundaryJustAboveEphemeral) {
+    auto [min_p, max_p] = ValidatePortRange(61000, 65000, 15000, 17000);
+    EXPECT_EQ(min_p, 61000);
+    EXPECT_EQ(max_p, 65000);
+}
```
[the remark this answers is not in the document]
Comment (jfeng18) [reply mooncake-transfer-engine/tests/config_test.cpp:765]: Already added in fee97d7: `LastBeforeEphemeral32767` test.

File: mooncake-transfer-engine/tests/config_test.cpp [hunk 4]
```diff
@@ -79,5 +81,253 @@ TEST_F(PkeyIndexEnvTest, EmptyStringKeepsDefault) {
...
+    EXPECT_EQ(config.log_level, google::ERROR);
+}
+
+TEST_F(LogLevelEnvTest, TraceEnablesTrace) {
+    ASSERT_EQ(::setenv("MC_LOG_LEVEL", "TRACE", 1), 0);
+    GlobalConfig config;
+    loadGlobalConfig(config);
+    EXPECT_EQ(config.log_level, google::INFO);
+    EXPECT_TRUE(config.trace);
+}
```
[the remark this answers is not in the document]
Comment (jfeng18) [reply mooncake-transfer-engine/tests/config_test.cpp:625]: Good suggestion. Will add.

**Comment by jfeng18** [comment]
CI note: the C++ build and unit tests both passed — step #14 `Build project` and step #20 `Test (in build env) with coverage` are green, so `config_test` (this PR's change) compiles and runs cleanly. `build (3.12)` only goes red later at step #26 `Build Python wheel` (which does not compile the test), and `build-wheel-cu13 (3.10)` failed on an sccache cache-backend DNS error before compilation. Both are environment-side; a re-run should clear them.

**Comment by jfeng18** [comment]
Hi @staryxchen @doujiang24 — this PR expands config env-var test coverage across 8 variables (39 cases, including full ValidatePortRange boundary matrices). It sat unnoticed since June, so I recently rebased it onto latest main: all 39 assertions re-verified against the current parsing logic, CI is green (27/27), and it merges cleanly. Pure test additions, no production code. Would appreciate a review when convenient. Thanks!

File: mooncake-transfer-engine/tests/config_test.cpp [hunk 5]
```diff
@@ -501,5 +503,332 @@ TEST_F(MaxWrEnvTest, OutOfRangeDoesNotSetFlag) {
...
+    EXPECT_EQ(max_p, 32767);
+}
+
+TEST(ValidatePortRangeTest, AboveMaxRejected) {
+    auto [min_p, max_p] = ValidatePortRange(61000, 65536, 15000, 17000);
+    EXPECT_EQ(min_p, 15000);
+    EXPECT_EQ(max_p, 17000);
+}
+
+// --- MC_MTU (valid values only; invalid causes exit()) ---
```
Comment (staryxchen) [inline mooncake-transfer-engine/tests/config_test.cpp]: Here it says invalid MC_MTU "causes exit()", but exit() was removed from config parsing in #2344 (before this PR was created). The current behavior is to LOG(ERROR) and ignore the invalid value (config.cpp:240-243).

Could you reword it to reflect that, e.g. // --- MC_MTU (valid values only; invalid is logged and ignored) ---? Since the invalid path no longer terminates the process, an InvalidIsIgnored test (asserting the default IBV_MTU_4096 is kept when e.g. MC_MTU=1500 is set) would also be safe to add now.

**Review by staryxchen — COMMENTED** [review]
Thanks for the thorough test expansion — verified locally that all 95 tests pass and assertions match the current parsing logic. Leaving one nit on this comment.

File: mooncake-transfer-engine/tests/config_test.cpp (hunk 5 above)
Comment (jfeng18) [reply mooncake-transfer-engine/tests/config_test.cpp]: Good catch — fixed in 682d3144: reworded the section comment, and added `MtuEnvTest.InvalidIsIgnored` asserting `mtu_length` stays `IBV_MTU_4096` with `MC_MTU=1500` (follows the existing fixture style of the group). Thanks also for verifying the full suite locally!

**Review by staryxchen — APPROVED** [review]
LGTM
